pymeasure/instruments/hp/hp437b.py

```diff
@@ -670,9 +669,8 @@ def resolution(self, resolution):
 
         """,
         validator=strict_discrete_set,
-        values={e: int(f.value) for (e, f) in zip(SensorType, SensorType)},
-        set_process=lambda v: SensorType(v) if isinstance(v, int) else v,
-        map_values=True,
+        values=[e for e in SensorType],
+        set_process=lambda v: int(v),
```

Comment:
That's the nice thing about `IntEnum`:
`"SE%dEN" % SensorType.SOME_SENSOR_TYPE` will use exactly the (int) value, so you do not need the `set_process` at all!